src/editor/map/map_context.cpp

```diff
@@ -122,21 +123,20 @@ map_context::map_context(const game_config_view& game_config, const std::string&
 	 * 0. Not a scenario or map file.
 	 *    0.1 File not found
 	 *    0.2 Map file empty
-	 *    0.3 No valid data
 	 * 1. It's a file containing only pure map data.
 	 *    * embedded_ = false
 	 *    * pure_map_ = true
-	 * 2. A scenario embedding the map
+	 * 2. The file contains an editor generated scenario file.
+	 *    * embedded_ = false
```

Comment:
Either here or at the declaration of `embedded_`, please document more clearly what this means.

The current doc for embedded_ is "Whether the map context refers to a map embedded in a scenario file.", I think clearer would be "True if the file that will be written to by clicking Save contains data that shouldn't be overwritten by the editor, in addition to data that can be generated by the editor. False if the complete file should be overwritten by the serialized version of the parts that the editor understands."

Reply:
I would say "Whether or not the map's terrain information is directly part of the scenario cfg file or exists as a separate '.map' file."

Reply:
It's being set to `false` here, but the terrain information is being saved in the .cfg file.

Reply:
Then: "Whether the map data is from a non-editor generated scenario file."

Reply:
But what does that mean? As in, if someone's reviewing a conditional statement based on this, how do they decide which behavior is the "should do this for true" and which is the "should do this for false"?

Reply:
Which part of the description isn't clear? "should do this for true" would be when dealing with a scenario that's hand-written, and "should do this for false" would be in the opposite case.